### PR TITLE
adds --partition option, fixes #623

### LINKED ISSUE #623: collect.py is not partition-aware
When the AWS account is in a partition other than `aws`, collect.py skips most collection tasks.  E.g., you get this when running in `aws-us-gov`:

```
* Getting ec2:describe-vpcs info
  Skipping region us-gov-west-1, as ec2 does not exist there                                                                                                                        
  Skipping region us-gov-east-1, as ec2 does not exist there
```

### PR DESCRIPTION
**Pull request by Cerebus**
* Adds `--partition` flag with default value `aws`
* Adds this value to `service.get_available_regions()` call

### CODE REVIEW

**Comment by Cerebus** [comment]
FYI, signed the CLA, but it's not updating.

**Comment by 0xdabbad00** [comment]
I've asked some folks to check up on why the CLA isn't updating.  Thank you for the PR!

File: commands/report.py [hunk 1]
```diff
@@ -221,17 +221,17 @@ def report(accounts, config, args):
         },
     ]
 
-    for account in accounts:
-        account = Account(None, account)
-        print("  - {}".format(account.name))
+    # for account in accounts:
```
Comment (0xdabbad00) [inline commands/report.py]: I can't merge this due these being commented out. Maybe a command-line flag could avoid these checks?

**Comment by 0xdabbad00** [comment]
@Cerebus See my comment about not being able to merge due to some of the functionality having been commented out.  Maybe a command-line flag could avoid this code, or maybe you could check if the iam data exists and skip if it doesn't?

**Comment by 0xdabbad00** [comment]
Closing this due to staleness as it breaks too many things.

**Pull request closed without merging**